### ISSUE TITLE
Feature Request: Tailscale HTTPS

### ISSUE THREAD

**Issue by tornupnegatives** [getumbrel/umbrel-apps#758]
Tailscale allows users to request an HTTPS certificate: https://tailscale.com/kb/1153/enabling-https/

It would be great if this functionality became supported for Umbrel, such that certain services like Vaultwarden work properly without Tor.

**Comment by aphex3k** [comment]
- Did you follow these steps? https://tailscale.com/kb/1153/enabling-https/#configure-https
- How is umbrel currently not supporting this?

**Comment by tornupnegatives** [comment]
Yes, I have. However, I do not see a way to run the `tailscale cert` command from Umbrel. Even if I SSH in, I would need to do it from the Docker container in which Tailscale is running, which I do not believe is possible. Ideally, the Tailscale Umbrel app would have a checkbox which triggers this command.

**Comment by aphex3k** [comment]
Did you check https://github.com/tailscale/tailscale/issues/5072?

**Comment by tornupnegatives** [comment]
No, but I will give it a try. Thanks.

Either way, there should be a user-friendly toggle within the Umbrel app itself.

**Comment by tornupnegatives** [comment]
HTTPS support is a major request over on the Umbrel community forums

**Comment by aphex3k** [comment]
> HTTPS support is a major request over on the Umbrel community forums

That might be the case, but you are "barking up the wrong tree". You can enable https on umbrel no problem after you SSH into your sever and use the appropriate docker command as described in the documentation. It works exactly as intended by the Tailscale developers.

**Comment by tornupnegatives** [comment]
Have you tried this? I actually cannot get it to work :/

**Comment by Knufle** [comment]
> > HTTPS support is a major request over on the Umbrel community forums
> 
> That might be the case, but you are "barking up the wrong tree". You can enable https on umbrel no problem after you SSH into your sever and use the appropriate docker command as described in the documentation. It works exactly as intended by the Tailscale developers.

What command would that be?

**Comment by FlyinPancake** [comment]
As of umbrelos 1.1 you can get a shell for your container and request the https cert from there

**Comment by tornupnegatives** [comment]
Per your recommendation, I have generated certificates using the `tailscale cert` command within the Tailscale container's shell. My Tailscale admin dashboard sees the certificate, but none of the Umbrel apps seem to be using it...

**Comment by sahilph** [comment]
> but none of the Umbrel apps seem to be using it...

They are not using it because, the internal containers are configured to be http only.

I have been working on getting https to work for all apps [More info here](https://github.com/getumbrel/umbrel/issues/546#issuecomment-2395428903)

It uses modified version of those containers which supports `https`. By default it creates self signed certificate, but if you intend to use tailscale exclusively, you can just, skip the "Trusting Root CA" part and replace the generated certificate files with the ones from tailscale.
